kolibri/auth/management/commands/fullfacilitysync.py

```diff
@@ -0,0 +1,98 @@
+import requests
+from django.utils.six.moves import input
+from kolibri.auth.constants.morango_scope_definitions import FULL_FACILITY
+from kolibri.auth.models import FacilityUser
+from kolibri.core.device.models import DevicePermissions, DeviceSettings
+from kolibri.core.device.utils import device_provisioned
+from kolibri.tasks.management.commands.base import AsyncCommand
+from morango.certificates import Certificate, Filter
+from morango.controller import MorangoProfileController
+
+
+class Command(AsyncCommand):
+
+    def add_arguments(self, parser):
+        parser.add_argument('--dataset-id', type=str)
+        parser.add_argument('--no-push', type=bool, default=False)
+        parser.add_argument('--no-pull', type=bool, default=False)
+        parser.add_argument('--base-url', type=str)
+        parser.add_argument('--username', type=str)
+        parser.add_argument('--password', type=str)
+
+    def handle_async(self, *args, **options):  # noqa: max-complexity=16
```

Comment:
Mhmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmmm

Reply:
I feel a bit stressed by the prospect that we'd be letting in this slip. We have coding standards for a reason, I think `# noqa` has to be used for a particular reason. I know we aren't perfect, but this code is really easy to improve, no?

The reason to avoid these long, complex functions is not just readability. But also test-ability. You'll find that writing tests is a lot easier by splitting functionality into smaller, well-defined chunks.

One other thing @ralphiee22 - if this is getting to long haired - is there any way to split this PR up so we can merge the PyPi dependency part and add this management command separately?

Reply:
Agree that it would be good to split this up, and avoid noqa. Wasn't pushing to do it right away because there's time pressure to get this in ASAP so that it gets manual testing exposure prior to tomorrow's launch, so we've been compiling a list of "immediate tech debt" to resolve after merging. But shouldn't be hard, and @ralphiee22's on it now.

Reply:
> is there any way to split this PR up so we can merge the PyPi dependency part and add this management command separately?

What was the goal? To be able to merge one before the other? The management command is actually one of the earliest commits in here, so that sounds like a hassle. Note that this management command is a "hidden feature" for the initial release, but we want to get it in there so we can start doing more field testing locally, and then with specific partners.